Source/WebGPU/WGSL/Overload.cpp

```diff
@@ -72,7 +72,7 @@ class OverloadResolver {
     bool unify(const AbstractType&, Type*);
     void assign(TypeVariable, Type*);
     Type* resolve(TypeVariable) const;
-    Type* materialize(const AbstractType&) const;
+    std::optional<Type*> materialize(const AbstractType&) const;
```

Comment:
Can't materializing a type create a new Type object? I don't quite understand how it only returns a raw pointer to a Type.